spectator/stateless_meters.h

```diff
@@ -46,7 +46,10 @@ class StatelessMeter {
     if (value_prefix_.empty()) {
       value_prefix_ = detail::create_prefix(*id_, Type());
     }
-    auto msg = absl::StrCat(value_prefix_, value);
+    auto msg = absl::StrFormat("%s%f", value_prefix_, value);
+    // remove trailing zeros and decimal points
+    msg.erase ( msg.find_last_not_of('0') + 1, std::string::npos );
```

Comment:
Could you just use `%.0f`? This seems like it could remove the wrong `0` if you ended up with a value like `1671641028.3`.

Reply:
In [absl::StrFormat](https://abseil.io/docs/cpp/guides/format), the `%.0f` pattern triggers rounding behavior, so we end up with results like the following:

```
Expected equality of these values:
  publisher.SentMessages()
    Which is: { "T:pt:0", "T:pt:0", "T:pt:5" }
  expected
    Which is: { "T:pt:0.042", "T:pt:0.0005", "T:pt:5" }
```

The [find_last_not_of](https://cplusplus.com/reference/string/string/find_last_not_of/) function searches the string from the end. This function can be used to remove trailing whitespace, as mentioned in the code example on the page.

The existing tests for meters have a variety of precision values specified, and these tests pass with this change.

Reply:
Hm, maybe the `%v` pattern does what we want here.